lib/db/lowlevel.go

```diff
@@ -822,6 +823,7 @@ func (db *Lowlevel) getMetaAndCheck(folder string) *metadataTracker {
 	var fixed int
 	fixed, err = db.checkLocalNeed([]byte(folder))
 	if err != nil {
+		err = fmt.Errorf("checking local need: %w", err)
```

Comment:
This set-a-local-var-and-return-nil-oops-we're-not-returning-after-all is some serious mindbending bullshit, though not a fault of this PR. :) But perhaps we can fix it, adding an error return here and making a `mustGetMetaAndCheck` that panics on that error? (Until we stop panicking on errors...)

Reply:
I already have a follow-up to https://github.com/syncthing/syncthing/pull/6925 that starts work on making db error handling less terrifying, and I hope suture lands soon so those pieces of the puzzle can also move into place. This really just is a tiny measure to make it at least it bit more transparently terrifying.

Reply:
Fine. Worse is better, or something.